### ISSUE TITLE
CARS FOR Cutsom ClassificationDataset. 

### ISSUE THREAD

**Issue by YTW0518** [huawei-noah/vega#214]
Dear prof.
        For 25*25*200 path images, where should this method modify the input settings?
Looking forward to your reply, thank you very very much!!!

**Comment by zhangjiajin** [comment]
https://github.com/huawei-noah/vega/blob/master/examples/features/custom_dataset/classification_dataset.yml

**Comment by YTW0518** [comment]
> @YTW0518
> 
> The following example may help you:
> 
> https://github.com/huawei-noah/vega/blob/master/examples/features/custom_dataset/classification_dataset.yml

Thank you for your timely reply. The main problem here is that my input data is not a three channel picture, but a hyperspectral picture with hundreds of channels. How should I set this?

**Comment by zhangjiajin** [comment]
@YTW0518 

You may need to modify the following function. Currently, this function can only process images of three channels.

<https://github.com/huawei-noah/vega/blob/47db340324cb39363de2ca54974c81a62c384b70/vega/datasets/common/cls_ds.py#L103>

```python
    def _load_image(self, image_file):
        img = PIL.Image.open(image_file)
        img = img.convert("RGB")
        return img
```

**Comment by YTW0518** [comment]
Thanks! ---- Replied Message ----Fromzhang ***@***.***>Date04/14/2022 15:43 ***@***.***> ***@***.******@***.***>SubjectRe: [huawei-noah/vega] CARS FOR Cutsom ClassificationDataset. (Issue #214)
Closed #214.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/huawei-noah/vega/issues/214#event-6432500374",
"url": "https://github.com/huawei-noah/vega/issues/214#event-6432500374",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
